pkg/eval/compile_value.go

```diff
@@ -377,10 +377,12 @@ func (cp *compiler) lambda(n *parse.Primary) valuesOp {
 		restArg       int = -1
 		optNames      []string
 		optDefaultOps []valuesOp
+		seenNames     map[string]bool
```

Comment:
This variable is only used inside the `if len(n.Elements) > 0 { ... }` block, so it's better kept inside that block. The other variables are declared here because they are used throughout the function.